File: problems/2587_rearrange_array_to_maximize_prefix_score/solution.py
from typing import List


class Solution:
    def maxScore(self, nums: List[int]) -> int:
        nums.sort(reverse=True)
        prefix = 0
        count = 0
        for n in nums:
            prefix += n
            if prefix > 0:
                count += 1
            else:
                break
        return count

File: problems/2587_rearrange_array_to_maximize_prefix_score/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int maxScore(vector<int>& nums) {
        sort(nums.rbegin(), nums.rend());
        long long sum = 0; int count = 0;
        for (int n : nums) { sum += n; if (sum > 0) count++; else break; }
        return count;
    }
};
